SimPEG/Mesh/View.py

```diff
@@ -358,22 +365,55 @@ def _plotImage2D(self, v, vType='CC', grid=False, view='real',
             U, V = self.r(v.reshape((self.nC,-1), order='F'), 'CC', 'CC', 'M')
             if clim is None:
                 uv = np.sqrt(U**2 + V**2)
-                clim = [uv.min(),uv.max()]
+                clim = [uv.min(), uv.max()]
 
             # Matplotlib seems to not support irregular
             # spaced vectors at the moment. So we will
             # Interpolate down to a regular mesh at the
             # smallest mesh size in this 2D slice.
-            nxi = int(self.hx.sum()/self.hx.min())
-            nyi = int(self.hy.sum()/self.hy.min())
-            tMi = self.__class__([np.ones(nxi)*self.hx.sum()/nxi,
-                                  np.ones(nyi)*self.hy.sum()/nyi], self.x0)
-            P = self.getInterpolationMat(tMi.gridCC,'CC',zerosOutside=True)
+            if range_x is not None:
+                dx = (range_x[1] - range_x[0])
+                nxi = int(dx/self.hx.min())
+                hx = np.ones(nxi)*dx/nxi
+                x0_x = range_x[0]
+            else:
+                nxi = int(self.hx.sum()/self.hx.min())
+                hx = np.ones(nxi)*self.hx.sum()/nxi
+                x0_x = self.x0[0]
+
+            if range_y is not None:
+                dy = (range_y[1] - range_y[0])
+                nyi = int(dy/self.hy.min())
+                hy = np.ones(nyi)*dy/nyi
+                x0_y = range_y[0]
+            else:
+                nyi = int(self.hy.sum()/self.hy.min())
+                hy = np.ones(nyi)*self.hy.sum()/nyi
+                x0_y = self.x0[1]
+
+            tMi = self.__class__([hx, hy], np.r_[x0_x, x0_y])
+            P = self.getInterpolationMat(tMi.gridCC, 'CC', zerosOutside=True)
+
             Ui = tMi.r(P*mkvc(U), 'CC', 'CC', 'M')
             Vi = tMi.r(P*mkvc(V), 'CC', 'CC', 'M')
             # End Interpolation
 
-            out += (ax.pcolormesh(self.vectorNx, self.vectorNy, np.sqrt(U**2+V**2).T, vmin=clim[0], vmax=clim[1], **pcolorOpts),)
+            x = self.vectorNx
+            y = self.vectorNy
+
+            ind_CCx = np.ones(self.vnC, dtype=bool)
+            ind_CCy = np.ones(self.vnC, dtype=bool)
+            if range_x is not None:
+                x = tMi.vectorNx
+
+            if range_y is not None:
+                y = tMi.vectorNy
+
+            if range_x is not None or range_y is not None:  # use interpolated values
```

Comment:
I am not sure we want to be doing this. If not, we just need a robust way to select cell centers for (U, V) and node locations for vector node locations. thoughts @rowanc1 ?